src/buildkite_test_collector/pytest_plugin/buildkite_plugin.py

```diff
@@ -48,3 +48,8 @@ def pytest_runtest_logreport(self, report):
 
             del self.in_flight[nodeid]
             self.payload = self.payload.push_test_data(test_data)
+
+    def save_payload(self, path):
+        """ Save payload into a json file """
```

Comment:
[nit] the comment looks like a better method name

```suggestion
    def save_payload_as_json(self, path):
        """ Save payload into a json file """
```